homeassistant/components/upnp/coordinator.py

```diff
@@ -35,12 +38,35 @@ def __init__(
             update_interval=update_interval,
         )
 
+    def register_entity(self, key: str, entity_id: str) -> Callable[[], None]:
+        """Register an entity."""
+        # self._entities.append(entity)
+        self._features_by_entity_id[key].add(entity_id)
+
+        def unregister_entity() -> None:
+            """Unregister entity."""
+            self._features_by_entity_id[key].remove(entity_id)
+
+            if not self._features_by_entity_id[key]:
+                del self._features_by_entity_id[key]
+
+        return unregister_entity
+
+    @property
+    def _entity_description_keys(self) -> list[str] | None:
+        """Return a list of entity description keys for which data is required."""
+        if not self._features_by_entity_id:
+            # Must be the first update, no entities attached/enabled yet.
+            return None
+
+        return list(self._features_by_entity_id.keys())
```

Comment:
Missed this one the first pass, but `keys()` can be dropped here since its the default